hydradx/model/amm/centralized_market.py

```diff
@@ -178,6 +182,8 @@ def swap(
                 for ask in self.order_book[(base, quote)].asks:
                     if ask[1] >= buy_tkns_remaining:
                         tkns_sold += buy_tkns_remaining * ask[0]
+                        if tkns_sold > agent.holdings[tkn_sell]:
+                            return self.fail_transaction('Agent does not have enough holdings to execute trade.')
```

Comment:
Do we fail Omnipool trades when agent doesn't have enough holdings to cover?

Reply:
yes